### ISSUE TITLE
[Bug]: mo hung on eks with 2*cn duaring oltp delete test 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9118]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-3dc68a0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

For daily regression test, mo was hung duaring oltp delte test.

<img width="1046" alt="image" src="https://user-images.githubusercontent.com/47846308/233783236-000c2c8f-be29-4841-a418-9570012b40aa.png">

<img width="1058" alt="image" src="https://user-images.githubusercontent.com/47846308/233783251-093b1b52-e850-4ae2-98f5-346e4c56a589.png">

<img width="948" alt="image" src="https://user-images.githubusercontent.com/47846308/233783270-8946d4c3-1d10-4662-b648-fa4419ee7904.png">

### Expected Behavior

The mo run normally

### Steps to Reproduce

```Markdown
1. git clone https://github.com/matrixorigin/mo-auto-test.git
2. modify .github/workflows/regression-on-distributed-mode.yml, 
  set JobName.default to oltp_delete_test
  set RunTimes to 10
3. make a pr to merge, and the job will automatically run the oltp delete queries for 10 times on eks
```

**Comment by zhangxu19830126** [comment]
[hung.tgz](https://github.com/matrixorigin/matrixone/files/11301372/hung.tgz)

**Comment by zhangxu19830126** [comment]
According to the stack log of hung, it is clear that push is blocked in:

**Comment by zhangxu19830126** [comment]
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/2995754/233785762-8444c328-1a2a-49d3-a694-175b7b7f97bb.png">

**Comment by m-schen** [comment]
明天确定一下是否是多cn环境下挂了一个cn导致的卡死。
用于接收log tail的协程是永远不会关闭，即使断连也会在协程中自行重连。

**Comment by aressu1985** [comment]
今天复现成功了，原因有两个：
1、Regression在执行过程中，导致DN OOM了，然后重新启动(这两天都是因为节点内存被耗尽，导致被节点被回收然后重建了一个)
<img width="839" alt="企业微信截图_4762d8b3-eaf5-4e37-ae76-d48cdb786ab6" src="https://user-images.githubusercontent.com/47846308/233895654-78c66629-48d3-4290-9b55-ba3e71e6aef0.png">

2、DN重新启动后，虽然DN状态正常，但是TP所有的QUERY都不能再正常工作，过一段时间就会报：SQL parser error: table "mo_account" does not exist

go routine的profile如下：
[mo-hung-dn-crash.txt](https://github.com/matrixorigin/matrixone/files/11306246/mo-hung-dn-crash.txt)

**Comment by iamlinjunhong** [comment]
wailting pr merge

**Comment by m-schen** [comment]
too confused. why hasn't this pr been merge..

**Comment by m-schen** [comment]
pr合不进去 无语

**Comment by m-schen** [comment]
pr已合并，主要解决的当dn挂掉后，cn触发重连时对错误信息处理不当导致的卡死问题。

**Comment by aressu1985** [comment]
fixed:c74710a
